dkg/pedersen/handler.go

```diff
@@ -26,13 +26,57 @@ import (
 // Allows to exit the loop.
 const recvTimeout = time.Second * 4
 
-// the time after which we expect new messages (deals or responses) to be
-// received.
-const retryTimeout = time.Millisecond * 200
+// the channel size used to stored buffered deals and responses. This arbitrary
+// value must be set according to the characteristic of the system: number of
+// nodes, networking, memory, etc...
+const chanSize = 200
 
 // constant used in the logs
 const newState = "new state"
 
+func newCryChan[T any](bufSize int) cryChan[T] {
+	llvl := zerolog.NoLevel
+	if os.Getenv("CRY_LVL") == "warn" {
+		llvl = zerolog.WarnLevel
+	}
+
+	return cryChan[T]{
+		c: make(chan T, bufSize),
+		log: dela.Logger.With().Str("role", "cry chan").Int("size", bufSize).
+			Logger().Level(llvl),
+	}
+}
+
+type cryChan[T any] struct {
+	c   chan T
+	log zerolog.Logger
+}
+
+func (c *cryChan[T]) push(e T) {
+	start := time.Now()
+	select {
+	case c.c <- e:
+	case <-time.After(time.Millisecond * 100):
```

Comment:
Since this is just a fail-safe, and 100ms is in the order of magnitude of a minor computer freeze, I'd consider increasing this to the range of seconds / tens of seconds.